### PR TITLE
Jakob/update gozk

### PR DESCRIPTION
**Pull request by ilikeorangutans**
Updates to the latest version of [github.com/Shopify/gozk](https://github.com/Shopify/gozk/commit/813187cc945353c7911040a8094941757f32ea5b) and exposes new methods on session to set automatic DNS refresh and querying current connection state and an options builder for cleaner session construction.

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -45,65 +47,56 @@ const (
 	// SessionFailed indicates that the session failed unrecoverably. This may mean incorrect credentials, or broken quorum,
 	// or a partition from the entire ZooKeeper cluster, or any other mode of absolute failure.
 	SessionFailed
+
+	DefaultRecvTimeout = 5 * time.Second
 )
 
-type ZKSession struct {
-	servers     string
+type SessionOpts struct {
```
Comment (awillock-shopify) [inline session/session.go]: Please extract to a separate file.

File: go.mod [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-go 1.17
+go 1.16
 
 require (
-	github.com/Shopify/gozk v0.0.0-20191029122301-2db116daa9ae
+	github.com/Shopify/gozk v0.0.0-20221028154422-a2c06c04a485
 	github.com/Shopify/toxiproxy/v2 v2.5.0
 )
 
-require launchpad.net/gocheck v0.0.0-20140225173054-000000000087 // indirect
```
Comment (Scalvando) [inline go.mod:10]: ❤️

**Review by Scalvando — APPROVED** [review]

File: session/session.go [hunk 3]
```diff
@@ -147,7 +135,7 @@ func (s *ZKSession) manage() {
 			case zookeeper.STATE_EXPIRED_SESSION:
 				s.log.Printf("gozk-recipes/session: got STATE_EXPIRED_SESSION for conn %+v", s.conn)
 				expired = true
-				conn, events, err := zookeeper.Redial(s.servers, s.recvTimeout, s.clientID)
+				conn, events, err := zookeeper.Redial(strings.Join(s.opts.servers, ","), s.opts.recvTimeout, s.opts.clientID)
```
Comment (awillock-shopify) [inline session/session.go:138]: Why the switch to the `strings.Join(...)` usage?

**Review by awillock-shopify — APPROVED** [review]

File: session/session.go (hunk 3 above)
Comment (ilikeorangutans) [reply session/session.go:138]: zookeeper.Redial takes comma separates lists, so I have to join the array.

Comment (awillock-shopify) [reply session/session.go:138]: Goddamnit. Reading comprehension fail. Thanks!

**Pull request merged**